net/arp/arp.h

```diff
@@ -81,8 +81,11 @@
 
 /* This is a helper pointer for accessing the contents of the IP header */
 
-#define ARPBUF    ((FAR struct arp_hdr_s *)&dev->d_buf[ETH_HDRLEN])
-#define ARPIPBUF  ((FAR struct arp_iphdr_s *)&dev->d_buf[ETH_HDRLEN])
+#define ARPBUF    ((FAR void *)(dev->d_iob ? \
```

Comment:
Why do we need `void *` cast?

Reply:
The type of io_data and d_buf are both uint8_t, in order to avoid too many cast convert in the code, so here use void * to cast directly